plugins/include/engine.inc

```diff
@@ -24,242 +24,1271 @@
+ *       other think functions like the pfn_think() forward before actually
+ *       blocking the think. To immediately block return PLUGIN_HANDLED_MAIN
+ *       instead.
+ *
+ * @param Touched   Entity classname to hook
+ * @param function  Name of callback function
+ *
+ * @noreturn
+ */
 native register_think(const Classname[], const function[]);
 
-/* NOTE: In old engine versions, this was not the case.  Values are now WINDOWS values.
- * You must pass with the windows offset (e.g. if 230 on windows, pass 230 no matter what) 
- * The module will automatically add +5 for Linux.
+/**
+ * Precaches an event file.
+ *
+ * @note The event type should always be 1.
+ * @note Contrary to the other precache_* natives, this can be used outside of
+ *       the plugin_precache() forward, e.g. in plugin_init() or plugin_cfg().
+ *       A bug in some clients makes this necessary, as plugin_precache() is
+ *       called before the mod has precached its own, default event files. This
+ *       can cause the event table to be misaligned on the client, leading to
+ *       visual and audio bugs that are hard to diagnose.
+ *
+ * @param type  Event type
+ * @param Name  Formatting rules, path to the event file
+ * @param ...   Variable number of formatting parameters
+ *
+ * @return      Unique cache id of the event
  */
-
-/* Precaches an event. */
 native precache_event(type, const Name[], any:...);
 
-/* set/get a user's speak flags */
+/**
+ * Sets the engine module speak flags on a client.
+ *
+ * @note For a list of available flags see the SPEAK_* constants in
+ *       engine_const.inc
+ *
+ * @param iIndex        Client index
+ * @param iSpeakFlags   New flags to set
+ *
+ * @noreturn
+ * @error               If the client index is not within the range of 1 to
+ *                      MaxClients, or the client is not connected, an error
+ *                      will be thrown.
+ */
 native set_speak(iIndex, iSpeakFlags);
+
+/**
+ * Returns the engine module speak flags currently set on a client.
+ *
+ * @note For a list of available flags see the SPEAK_* constants in
+ *       engine_const.inc
+ *
+ * @param iIndex    Client index
+ *
+ * @return          Client speak flags
+ * @error           If the client index is not within the range of 1 to
+ *                  MaxClients, or the client is not connected, an error will be
+ *                  thrown.
+ */
 native get_speak(iIndex);
 
-/* Drops an entity to the floor (work?) */
+/**
+ * Uses the DROP_TO_FLOOR() engine function on an entity, which attempts to put
+ * it down on the floor.
+ *
+ * @note This engine function traces 256 units straight downwards from the
+ *       entity origin. If the trace hits the floor, the origin is updated to
+ *       the end position of the trace, FL_ONGROUND is added to the flags and
+ *       EV_ENT_groundentity is updated. When the trace does not hit anything or
+ *       the entity would be stuck inside something, the process fails.
```

Comment:
Yes. It runs the trace and if the check fails (nothing hit or trace stuck inside something) it returns 0 and does nothing at all. I can reword that to make it more clear.